### DISCUSSION TITLE
NextJS - Support for dynamic routes

### DISCUSSION THREAD

**Discussion by yarrichar** [markdoc/markdoc#360] [Q&A] [answered]
Hi,

I'm making a nextjs site and am wanting to pull my markdoc content from a database (i.e. not a static .md file sitting under `/pages/`). 

This is nearly working, but when I do the following in `/pages/[id]/index.tsx` I'm not getting any of my schema definitions applied (i.e. all the stuff that lives under `/markdoc/`):
```javascript
const transformed = Markdoc.transform(Markdoc.parse(content));
    
return <div>
    {Markdoc.renderers.react(transformed, React)}
</div>
```

The .md files that live under `/pages/` work fine - this also includes a custom tag I made.

If anyone knows how to do this some help would be appreciated.

Sorry if this is a silly question - new to both markdoc and nextjs.

**Comment by mfix-stripe** [comment, marked as answer]
@yarrichar there might be a few things going on, but at the very least you need to pass a `components` config to `Markdoc.renderers.react`: https://github.com/markdoc/docs/blob/main/components/Sandbox.js#L428-L430

Without a more thorough example, I don't think I'll be able to help any more than that /:
